_field-types/index.md

```diff
@@ -14,38 +14,57 @@ redirect_from:
 
-#### Table of contents
-1. TOC
-{:toc}
-
-
----
 ## Dynamic mapping
 
 When you index a document, OpenSearch adds fields automatically with dynamic mapping. You can also explicitly add fields to an index mapping.
 
-#### Dynamic mapping types
+### Dynamic mapping types
 
 Type | Description
 :--- | :---
 null | A `null` field can't be indexed or searched. When a field is set to null, OpenSearch behaves as if that field has no values.
-boolean | OpenSearch accepts `true` and `false` as boolean values. An empty string is equal to `false.`
+boolean | OpenSearch accepts `true` and `false` as Boolean values. An empty string is equal to `false.`
 float | A single-precision 32-bit floating point number.
 double | A double-precision 64-bit floating point number.
 integer | A signed 32-bit number.
 object | Objects are standard JSON objects, which can have fields and mappings of their own. For example, a `movies` object can have additional properties such as `title`, `year`, and `director`.
-array | Arrays in OpenSearch can only store values of one type, such as an array of just integers or strings. Empty arrays are treated as though they are fields with no values.
+array | Arrays in OpenSearch can only store values of one type, such as an array of only integers or strings. Empty arrays are treated as though they are fields with no values.
 text | A string sequence of characters that represent full-text values.
 keyword | A string sequence of structured characters, such as an email address or ZIP code.
 date detection string | Enabled by default, if new string fields match a date's format, then the string is processed as a `date` field. For example, `date: "2012/03/11"` is processed as a date.
 numeric detection string | If disabled, OpenSearch may automatically process numeric values as strings when they should be processed as numbers. When enabled, OpenSearch can process strings into `long`, `integer`, `short`, `byte`, `double`, `float`, `half_float`, `scaled_float`, and `unsigned_long`. Default is disabled.
 
+### Dynamic templates
+
+Dynamic templates are used to define custom mappings for dynamically added fields based on data type, field name, or field path. They allow you to define a flexible schema for your data, which can automatically adapt to changes in the structure or format of the input data.
+
+The syntax for defining a dynamic mapping template in OpenSearch looks like the following:
+
+```json
+  "dynamic_templates": [
+    {
+      "template_name": { 
+        ...  match conditions ... 
+        "mapping": { ... } 
+      }
+    },
+    ...
+  ]
+```
+{% include copy-curl.html %}
+
+Note the following: 
+
+- The template name can be any string value. 
+- The match conditions can include any of the following: `match_mapping_type`, `match`, `match_pattern`, `unmatch`, `path_match`, or `path_unmatch`. 
```

Comment:
The supported params for matching conditions with `dynamic_templates` are:

`match_mapping_type`: Detected JSON type based on which mapping should be applied. (e.g. string, long, double, object, binary, boolean, date)
`match`: Regex for matching field name and apply mapping
`unmatch`: Regex to verify and not apply mapping if the field name satisfies the provided regex
`match_pattern`: Allowed values are `regex` and `simple`. Based on this, it applies the regex matching. Default is `simple`
`path_match`: Allows for nested path matching along with regex
`path_unmatch`: Allows for nested path matching along with regex
`mapping`: Mapping to be applied

Reply:
Added.